pkg/crypto/dh_test.go

```
@@ -0,0 +1,87 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package crypto_test
+
+import (
+	"bytes"
+	"crypto/ecdsa"
+	"crypto/rand"
+	"encoding/hex"
+	"io"
+	"testing"
+
+	"github.com/btcsuite/btcd/btcec"
+	"github.com/ethersphere/bee/pkg/crypto"
+)
+
+func TestECDHCorrect(t *testing.T) {
+	key0, err := crypto.GenerateSecp256k1Key()
+	if err != nil {
+		t.Fatal(err)
+	}
+	dh0 := crypto.NewDH(key0)
+
+	key1, err := crypto.GenerateSecp256k1Key()
+	if err != nil {
+		t.Fatal(err)
+	}
+	dh1 := crypto.NewDH(key1)
+
+	salt := make([]byte, 32)
+	if _, err := io.ReadFull(rand.Reader, salt); err != nil {
+		panic("reading from crypto/rand failed: " + err.Error())
+	}
+	
+	sk0, err := dh0.SharedKey(&key1.PublicKey, salt)
+	if err != nil {
+		t.Fatal(err)
+	}
+	sk1, err := dh1.SharedKey(&key0.PublicKey, salt)
```

Error is not checked.